src/components/protocol_handler/src/protocol_packet.cc

```diff
@@ -233,6 +249,9 @@ RESULT_CODE ProtocolPacket::ProtocolHeaderValidator::validate(
   // on used protocol version and service type
   size_t payload_size = MAXIMUM_FRAME_DATA_V2_SIZE;
   // Protocol version shall be from 1 to 4
+  if (max_protocol_version_supported_ < header.version) {
```

Comment:
@SKobziev Seems that this check is redundant, because we will go to default part of switch construction